src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -473,9 +467,11 @@ internal static unsafe void CreateAnonymousPipe(HandleInheritability inheritabil
                 s.SendBufferSize = outBufferSize;
             }
 
-            if (inheritability != HandleInheritability.Inheritable)
+            // Sockets are created with O_CLOEXEC.  If inheritability has been requested, we need to unset the flag.
+            if (inheritability == HandleInheritability.Inheritable &&
+                Interop.Sys.Fcntl.SetFD(s.SafeHandle, 0) == -1)
             {
-                Interop.Sys.Fcntl.SetCloseOnExec(pipeHandle); // ignore failures, best-effort attempt
+                throw Interop.GetExceptionForIoErrno(Interop.Sys.GetLastErrorInfo());
```

Comment:
Maybe we should have a parameterless overload to `GetExceptionForIoErrno` that gets `GetLastErrorInfo()` for itself.

Reply:
We could, though it looks like this only occurs three times across corefx right now.